packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -1871,6 +1871,13 @@ def __init__(
         logger: logging.Logger,
     ):
         """Initialize the AbciApp."""
+
+        if not hasattr(self.initial_round_cls, "synchronized_data_class"):
+            logger.warning(f"No `synchronized_data_class` set on {self}")
```

Comment:
why are we not raising?

Reply:
We could but that would break things. Many skills don't use this (e.g. `registration_abci`, `reset_pause_abci`). Also this can be abstracted away entirely when we get to the refactoring, it has been added to that list on @Adamantios 's request.